quickstart/backend/src/main/java/com/digitalasset/quickstart/service/AppInstallRequestsApiImpl.java

```diff
@@ -42,128 +51,199 @@ public AppInstallRequestsApiImpl(
-                            new quickstart_licensing.licensing.util.Metadata(appInstallRequestAccept.getMeta().getData())
-                    );
-
-            return ledger.exerciseAndGetResult(providerParty, contract.contractId, choice, commandId)
-                    .thenApply(appInstallContractId -> {
-                        // Build response AppInstall object
-                        AppInstall appInstallResponse = new AppInstall();
-                        appInstallResponse.setDso(contract.payload.getDso.getParty);
-                        appInstallResponse.setProvider(contract.payload.getProvider.getParty);
-                        appInstallResponse.setUser(contract.payload.getUser.getParty);
-                        // Set meta to the installMeta provided in the choice
-                        appInstallResponse.setMeta(appInstallRequestAccept.getInstallMeta());
-                        appInstallResponse.setNumLicensesCreated(0);
-
-                        return ResponseEntity.ok(appInstallResponse);
-                    }).exceptionally(ex -> {
-                        logger.error("Error accepting AppInstallRequest", ex);
-                        return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).build();
-                    });
-        }).exceptionally(ex -> {
-            logger.error("Error fetching AppInstallRequest with contractId: {}", contractId, ex);
-            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).build();
-        });
+        Span methodSpan = Span.current();
+        Context parentContext = Context.current();
+
+        methodSpan.addEvent("Starting acceptAppInstallRequest");
+        methodSpan.setAttribute("contractId", contractId);
+        methodSpan.setAttribute("commandId", commandId);
+        logger.info("acceptAppInstallRequest: contractId='{}', commandId='{}'", contractId, commandId);
+
+        // Capture the provider party asynchronously
+        return authenticatedPartyService.getPartyOrFail()
+                .thenCompose(providerParty ->
+                        damlRepository.findAppInstallRequestById(contractId)
+                                .thenCompose(contract -> {
+                                    methodSpan.addEvent("Fetched contract, exercising AppInstallRequest_Accept choice");
+
+                                    quickstart_licensing.licensing.appinstall.AppInstallRequest.AppInstallRequest_Accept choice =
+                                            new quickstart_licensing.licensing.appinstall.AppInstallRequest.AppInstallRequest_Accept(
+                                                    new quickstart_licensing.licensing.util.Metadata(appInstallRequestAccept.getInstallMeta().getData()),
+                                                    new quickstart_licensing.licensing.util.Metadata(appInstallRequestAccept.getMeta().getData())
+                                            );
+
```

Comment:
This is where the business logic invokes the ledger API. There should be a log at DEBUG of this transition with a suitable selection of the business properties, and the contractId + commandId for correlation to with the ledgerAPI logs.